src/darwin/Framework/CHIP/MTRDeviceControllerDataStore.mm

```diff
@@ -699,6 +718,45 @@ - (void)clearStoredClusterDataForNodeID:(NSNumber *)nodeID
     });
 }
 
+- (void)clearStoredClusterDataForNodeID:(NSNumber *)nodeID endpointID:(NSNumber *)endpointID clusterID:(NSNumber *)clusterID
+{
+    dispatch_async(_storageDelegateQueue, ^{
+        BOOL success = [self _deleteClusterDataForNodeID:nodeID endpointID:endpointID clusterID:clusterID];
+        if (!success) {
+            MTR_LOG_ERROR("clearStoredClusterDataForNodeID: _deleteClusterDataForNodeID failed for node 0x%016llX endpoint %u cluster 0x%08lX", nodeID.unsignedLongLongValue, endpointID.unsignedShortValue, clusterID.unsignedLongValue);
+            return;
+        }
+
+        NSArray<NSNumber *> * clusterIndex = [self _fetchClusterIndexForNodeID:nodeID endpointID:endpointID];
```

Comment:
Again, shouldn't the index update happen first, before we remove things?  Otherwise we can remove things, fail to update index, and be in an inconsistent state.

Please double-check with @jtung-apple to be sure.

Reply:
fixed. i checked existing code written by Jeff to do some other pruning but yes his code deletes the cluster index first. will double check once he's back tommorrow